patches/master_patch.patch

```diff
@@ -1324,41 +1338,52 @@ index 829b2dc1c65e9c2d5226592b3b33c16ee4070b8b..a28adcf4478dddd5089271b7a4755b3b
    notify_browser_of_print_failure_ = false;
    if (!success)
      LOG(ERROR) << "Failure in OnPrintingDone";
-@@ -1566,8 +1572,8 @@ void PrintWebViewHelper::DidFinishPrinting(PrintingResult result) {
+@@ -1570,8 +1576,8 @@ void PrintWebViewHelper::DidFinishPrinting(PrintingResult result) {
        }
        break;
  
 -#if BUILDFLAG(ENABLE_PRINT_PREVIEW)
      case FAIL_PREVIEW:
 +#if 0
-       int cookie =
-           print_pages_params_ ? print_pages_params_->params.document_cookie : 0;
-       if (notify_browser_of_print_failure_) {
-@@ -1576,9 +1582,13 @@ void PrintWebViewHelper::DidFinishPrinting(PrintingResult result) {
-       } else {
-         Send(new PrintHostMsg_PrintPreviewCancelled(routing_id(), cookie));
+       if (!is_print_ready_metafile_sent_) {
+         if (notify_browser_of_print_failure_) {
+           LOG(ERROR) << "CreatePreviewDocument failed";
+@@ -1581,13 +1587,24 @@ void PrintWebViewHelper::DidFinishPrinting(PrintingResult result) {
+         }
        }
--      print_preview_context_.Failed(notify_browser_of_print_failure_);
+       print_preview_context_.Failed(notify_browser_of_print_failure_);
 +#endif  // 0
 +      LOG(ERROR) << "PREVIEW FAILED.";
 +      if (print_pages_params_) {
 +        Send(new PrintHostMsg_PrintPreviewFailed(routing_id(),
 +                 print_pages_params_->params.document_cookie));
 +      }
        break;
+     case INVALID_SETTINGS:
++#if 0
+       Send(new PrintHostMsg_PrintPreviewInvalidPrinterSettings(routing_id(),
+                                                                cookie));
+-      print_preview_context_.Failed(false);
++#endif // 0
++      LOG(ERROR) << "PREVIEW FAILED.";
```

Comment:
this looks like a dup of the code above?

Reply:
Sure. I'm also addressing it.